runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/utils/SerializedPipelineOptions.java

```diff
@@ -52,12 +53,13 @@ public PipelineOptions getPipelineOptions() {
     if (pipelineOptions == null) {
       try {
         pipelineOptions = new ObjectMapper().readValue(serializedOptions, PipelineOptions.class);
+
+        IOChannelUtils.registerIOFactoriesAllowOverride(pipelineOptions);
```

Comment:
This may have an issue in the following scenario (if realistic):
- Two different pipelines submitted to a cluster, each with their own pipeline options.
- The same worker runs tasks from both pipelines at the same time, in the same process.
- If that can happen, the second registration may override the first; leaking pipeline options from the second pipeline to the first.

Even if this is possible, I'd prefer to submit to unblock the scenario, and we can think about how to fix this in the future.